packages/testing/src/execution_testing/cli/pytest_commands/plugins/shared/benchmarking.py

```diff
@@ -54,192 +56,229 @@ def pytest_configure(config: pytest.Config) -> None:
+            )
+        return getattr(config, cls.config_field)
+
+    @abstractmethod
+    def get_test_parameters(self, test_name: str) -> list[ParameterSet]:
+        """Get the parameters list for a given test."""
+        pass
+
+    def parametrize(self, metafunc: pytest.Metafunc) -> None:
+        """Parametrize a test."""
+        if self.parameter_name in metafunc.fixturenames:
+            test_name = metafunc.function.__name__
+            metafunc.parametrize(
+                self.parameter_name,
+                self.get_test_parameters(test_name),
+                scope="function",
+            )
+
+
+class GasBenchmarkValues(RootModel, BenchmarkParametrizer):
+    """Gas benchmark values configuration object."""
+
+    root: List[int]
+
+    flag: ClassVar[str] = "--gas-benchmark-values"
+    config_field: ClassVar[str] = "_gas_benchmark_values_config"
+    parameter_name: ClassVar[str] = "gas_benchmark_value"
+
+    @classmethod
+    def from_parameter_value(
+        cls, config: pytest.Config, value: str
+    ) -> Self | None:
+        """Given the parameter value and config, return the expected object."""
+        return cls.model_validate(value.split(","))
+
+    def get_test_parameters(self, test_name: str) -> list[ParameterSet]:
+        """Get benchmark values. All tests have the same list."""
+        return [
+            pytest.param(
+                gas_value * 1_000_000,
+                id=f"benchmark-gas-value_{gas_value}M",
+            )
+            for gas_value in self.root
+        ]
+
+
+class OpcodeCountsConfig(BaseModel, BenchmarkParametrizer):
+    """Opcode counts configuration object."""
+
+    scenario_configs: Dict[str, List[int]] = Field(default_factory=dict)
+    default_counts: List[int] = Field(default_factory=lambda: [1])
+
+    default_config_file_name: ClassVar[str] = ".fixed_opcode_counts.json"
+    flag: ClassVar[str] = "--fixed-opcode-count"
+    config_field: ClassVar[str] = "_opcode_counts_config"
+    parameter_name: ClassVar[str] = "fixed_opcode_count"
+
+    @classmethod
+    def from_parameter_value(
+        cls, config: pytest.Config, value: str
+    ) -> Self | None:
+        """Given the parameter value and config, return the expected object."""
+        if value == "":
+            default_file = Path(config.rootpath) / cls.default_config_file_name
+            if default_file.exists():
+                return cls.model_validate_json(default_file.read_bytes())
+            else:
+                warnings.warn(
+                    "--fixed-opcode-count was provided without a value, but "
+                    f"{cls.default_config_file_name} was not found. "
+                    "Run 'uv run benchmark_parser' to generate it, or provide "
+                    "explicit values (e.g., --fixed-opcode-count 1,10,100).",
+                    UserWarning,
+                    stacklevel=1,
+                )
+                return cls()
```

Comment:
I think an explicit fail is more obvious behavior than a warning and running with an unexpected default value.
```suggestion
                pytest.UsageError(
                    "--fixed-opcode-count was provided without a value, but "
                    f"{cls.default_config_file_name} was not found. "
                    "Run 'uv run benchmark_parser' to generate it, or provide "
                      "explicit values (e.g., --fixed-opcode-count 1,10,100)."
                )
```